### ISSUE TITLE
Unload leaks the house-consumer bus listeners and the pulse task: reload double-fires lights and SFX

### ISSUE THREAD

**Issue by mholzi** [mholzi/quizify#605]
## What happens

`async_unload_entry` (`__init__.py:600-636`) flushes question stats, cancels game tasks, removes the panel and pops `hass.data[DOMAIN]` — but never calls `detach()` on the five house consumers.

The options-change path right next to it does exactly that:

```python
# __init__.py:506-515  (_update_listener)
if pl is not None:  pl.detach()
if tts is not None: tts.detach()
if lm is not None:  lm.detach()
if ev is not None:  ev.detach()
if sfx is not None: sfx.detach()
```

The asymmetry is the bug. `QuizifyPartyLights.attach_events` holds five `hass.bus.async_listen` subscriptions (`lights.py:395-403`) and `QuizifySoundEffects.attach_events` holds three (`sound_effects.py:241-245`). The lights pulse task (`hass.async_create_task`, `lights.py:567`) is cancelled only inside `detach()`. Popping `hass.data` does not unsubscribe anything — the bus keeps the old instances alive.

## Symptom

Trigger: a config-entry reload with party lights or SFX configured. That means the Integrations UI "Reload" button **and every HACS update**.

- After N reloads, every `quizify_*` event drives `light.turn_on` / `media_player.play_media` N+1 times — double flashes, overlapping stings.
- After *removing* the integration, the lights and SFX keep reacting to any leftover events until Home Assistant restarts.

`tests/test_init_313.py::test_unload_entry_tears_down` asserts only `hass.data` and entity teardown, so this path is uncovered.

## Suggested fix

Call `detach()` on all five consumers in `async_unload_entry`, in the same sequence `_update_listener` already uses, before popping `hass.data`. Extend the unload test to assert the bus listener count returns to its pre-setup value — asserting `hass.data` is empty is what let this through.

*Found by an automated review pass, verified against the source.*
